src/interactableComponents/drawer/index.js

```diff
@@ -0,0 +1,469 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Animated, View, TouchableOpacity, TouchableHighlight} from 'react-native';
+import Interactable from 'react-native-interactable';
+import {BaseComponent, Constants, Colors, Typography} from '../../../src';
+
+
+const DEFAULT_HEIGHT = 72;
+const MIN_LEFT_MARGIN = 28;
+const DEFAULT_ICON_SIZE = 24;
+const ITEM_BG = Colors.blue30;
+const ITEM_PADDING = 12;
+const BLEED = 15;
+const ITEM_PROP_TYPES = {
+  width: PropTypes.number,
+  background: PropTypes.string,
+  text: PropTypes.string,
+  icon: PropTypes.number,
+  onPress: PropTypes.func,
+};
+
+/**
+ * @description: Interactable Drawer component
+ * @extendslink: 
+ */
+export default class Drawer extends BaseComponent {
+  static displayName = 'Drawer';
+
+  static propTypes = {
+    /**
+     * The drawer's height
+     */
+    height: PropTypes.number.isRequired,
```

Comment:
The `height` & `width` props, in my opinion, should be derived from the wrapped component. 
I know we talked about it already. But, I saw other libraries and usually, you don't send these props. 

I know we might need them for all kind of calculations, but, I think the internal behavior define in our guidelines  should be part of the internal (private) uilib code. 

This will make this component way more generic and simple.

Reply:
Fixed!